aes128.cpp

```diff
@@ -106,6 +106,247 @@ void cipher::aes128::m_decrypt_block(byte_block& block) const {
     m_xor_blocks(block, m_key_schedule[0]);
 }
 
+cipher::aes128::byte_block cipher::aes128::generate_initialization_vector() const
+{
+    byte_block initialization_vector;
+    srand(time(NULL));
```

Comment:
Лучше использовать библиотеку `<random>`, т.к. она обеспечивает лучшее, по сравнению с `rand()`, распределение + есть возможность задать истинно рандомный seed с помощью `std::random_device`
Должно быть что-то типа:
```suggestion
    std::mt19937 gen(std::random_device{}());
    std::uniform_int_distibution<uint8_t> random_symbol(0, 255);
```
`random_symbol(gen)` вернет рандомное значение

Reply:
Исправила